### ISSUE TITLE
Check for `if` conditions that are always truthy

### ISSUE THREAD

**Issue by kkpattern** [astral-sh/ty#1073]
I wish ty can check for comparing coroutine in if statement. Given the following code:

```python
async def check_foo() -> bool:
    return True

async def main():
    if check_foo():
        print("hello")
```

we almost never want to compare a coroutine in if statement because it will always be true. What we want is actually:

```python
async def main():
    if await check_foo():
        print("hello")
```

I wonder if this should be a ty rule or ruff rule? I don't know if ruff knows the `check_foo` is a coroutine during the check. If it should be a ruff rule I will move this issue to the ruff repo.

Thanks.

**Comment by carljm** [comment]
Ruff could check this, but only in a very incomplete way; if `check_foo()` came from a different module, it would not know that it is an async function.

Long-term our plan is to build type-powered linting on top of ty, but at the moment we are still focused on getting the type-checker feature set complete.

I do think this is an ideal example of a type-powered lint rule, and we should definitely build it, when we are at that point of adding such rules.

**Comment by kkpattern** [comment]
Awesome!

**Comment by charliermarsh** [comment]
This is now reported by `redundant-condition` (and suggests `await`).
